ara/utils.py

```diff
@@ -98,6 +88,51 @@ def default_data():
     return data
 
 
+def fields_from_iter(fields, items, xforms=None):
+    '''Returns column headers and data for use by
+    `cliff.lister.Lister`.  In this function and in
+    `fields_from_object`, fields are specified as a list of
+    `(column_name, object_path)` tuples.  The `object_path` can be
+    omitted if it can be inferred from the column name by converting
+    the name to lowercase and converting ' ' to '_'.  For example:
+
+        fields = (
+            ('ID',),
+            ('Name',),
+            ('Playbook',),
+        )
+
+    The `xforms` parameter is a dictionary maps column names to
+    callables that will be used to format the corresponding value.
+    For example:
+
+        xforms = {
+            'Playbook': lambda p: p.name,
+        }
+    '''
+
+    xforms = xforms or {}
+
+    return (zip(*fields)[0], [
+        [xform(v) for v, xform in
+         [(get_field_attr(item, f[0]), f[1]) for f in
+          [(field, xforms.get(field[0], lambda x: x)) for field in fields]]]
+        for item in items])
+
+
+def fields_from_object(fields, obj, xforms=None):
```

Comment:
Now, that said, we're not actually *modifying* `xforms` inside the function, but for those reasons I generally avoid using dictionaries or lists for default values.